src/float.cr

```diff
@@ -148,8 +148,17 @@ struct Float32
     Printer.print(self, io)
   end
 
+  private def hash_normalize
+    t = to_i32
+    t == self ? t : unsafe_as(Int32)
```

Comment:
Does #4581 fix hashing of `Float32|Float64` ? If so, then it could be better to apply.
Do you want I include it to my patch, or they will be applied in some order?